Common/Utility/DocDbRestUtility.cs

```diff
@@ -267,6 +270,26 @@ public DocDbRestUtility(IConfigurationProvider configProvider)
             return JObject.Parse(response);
         }
 
+        public async Task<DeviceND> SaveNewDocumentAsyncND(DeviceND document)
+        {
+            if (document == null)
+            {
+                throw new ArgumentNullException("document");
+            }
+
+            string endpoint = string.Format("{0}dbs/{1}/colls/{2}/docs", _docDbEndpoint, _dbId, _collectionId);
+            if (document.id == null)
+            {
+                document.id = Guid.NewGuid().ToString();
+            }
+
+            string response = await PerformRestCallAsync(endpoint, POST_VERB, DocDbResourceType.Document, _collectionId, document.ToString());
+
+            var dynamicDevice = JObject.Parse(response);
+            DeviceND device = TypeMapper.Get().map<DeviceND>(dynamicDevice);
```

Comment:
let's keep the mapper until everyone validates their changes. Will change to your suggestion when we're doing cleanup